src/search/SearchMessagesCard.js

```diff
@@ -52,6 +52,9 @@ export default class SearchMessagesCard extends PureComponent<Props> {
           disableFetching
           showMessagePlaceholders={false}
           suppressTypingIndicator
+          // TODO: handle editing a message from the search results,
+          // or don't let callers pass this.
+          startEditMessage={() => undefined}
```

Comment:
What do you have in mind for "don't let callers pass this"? The two natural options I see here are
* handle editing a message from the search results, or
* make this optional.

Probably we don't want to try to handle editing from here, so that'd point to the latter. We already have this where we construct the action sheet:
```js
  if (
    message.sender_id === ownUser.user_id
    // Our "edit message" UI only works in certain kinds of narrows.
    && (isStreamOrTopicNarrow(narrow) || isPmNarrow(narrow))
  ) {
    buttons.push(editMessage);
  }
```

Perhaps that filter on the narrow could move up to `ChatScreen`, and control whether we pass a non-void `startEditMessage` callback at all? (And here in search, just never pass it.) Then in the action sheet we could just check if the callback is truthy.

Reply:
> don't let callers pass this

I think I meant to say "don't require callers to pass this".

Reply:
> Perhaps that filter on the narrow could move up to `ChatScreen`, and control whether we pass a non-void `startEditMessage` callback at all? (And here in search, just never pass it.) Then in the action sheet we could just check if the callback is truthy.

Yeah; just this week, it's come up that it'd be nice to do this conditional in just one place: https://github.com/zulip/zulip-mobile/pull/4799#discussion_r651298734.

To check if the callback is truthy, in the action sheet, are we OK with `constructMessageActionButtons` taking another param? Currently it looks like this:

```js
export const constructMessageActionButtons = ({
  backgroundData: { ownUser, flags },
  message,
  narrow,
}: {
  backgroundData: $ReadOnly<{
    ownUser: User,
    flags: FlagsState,
    ...
  }>,
  message: Message,
  narrow: Narrow,
}): Button<MessageArgs>[] => {
  // ...
}
```

Reply:
Hmm, yeah, it is a little awkward to pass it down there (through `constructNonHeaderActionButtons`, too. But I think it's OK.